src/packages/engine-core/src/NodeEngine.ts

```diff
@@ -542,6 +542,7 @@ ${chalk.dim("In case we're mistaken, please report this to us 🙏.")}`)
       }
     }
 
+    // TODO What is this and why?
     if (this.datasources) {
       env.OVERWRITE_DATASOURCES = this.printDatasources()
     }
```

Comment:
It's for overriding the datasource from client constructor https://github.com/prisma/prisma/issues/2443#issuecomment-630679118
```
import { PrismaClient } from '@prisma/client'

const client1 = new PrismaClient({ datasources: { db: { url: 'postgres://localhost/db1' }} })
```

Reply:
And then `printDatasources()` is used to actually get it, and then it is submitted to the Engine via `env.OVERWRITE_DATASOURCES`? Do we have an equivalent for this in the Node-API Engine?

Reply:
Looks like yes https://github.com/prisma/prisma/blob/cc729c77b95c94bf9a57fe89aee6c3c043def278/src/packages/engine-core/src/NAPIEngine.ts#L92-L94